lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -267,10 +275,12 @@ defmodule Mix.Compilers.Elixir do
         binary: binary
       )
 
+    source(size: size) = List.keyfind(sources, source, source(:source))
+
     new_source =
       source(
         source: source,
-        size: :filelib.file_size(source),
+        size: size,
```

Comment:
Since we now have the correct file size information inside the Process dictionary, I think we probably don't need to access the file system here anymore.